food-delivery/packages/workflows/order.ts

```diff
@@ -0,0 +1,98 @@
+import {
+  defineSignal,
+  defineQuery,
+  setHandler,
+  condition,
+  ApplicationFailure,
+  sleep,
+  proxyActivities,
+} from '@temporalio/workflow'
+import { errorMessage, getProductById, Product } from 'common'
+import type * as activities from 'activities'
+
+type OrderState = 'Charging card' | 'Paid' | 'Picked up' | 'Delivered'
+
+export interface OrderStatus {
+  productId: number
+  state: OrderState
+  deliveredAt?: Date
+}
+
+export const pickedUp = defineSignal('pickedUp')
+export const delivered = defineSignal('delivered')
+export const getStatus = defineQuery<OrderStatus>('getStatus')
+
+const { chargeCustomer, refundOrder, sendPushNotification } = proxyActivities<typeof activities>({
+  startToCloseTimeout: '1m',
+  retry: {
+    maximumInterval: '5s',
```

Comment:
Why? Is this just for demo purposes?
In a real world example you wouldn't want to limit the max interval like this to avoid overloading the upstream services.

Reply:
yeah, was waiting too long for success sometimes. will add comment